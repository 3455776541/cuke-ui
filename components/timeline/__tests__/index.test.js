import assert from "power-assert";
import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Timeline from "../index";

describe("<Timeline/>", () => {
  it("should render Timeline", () => {
    const wrapper = render(
      <Timeline>
        <Timeline.Item>三天前 lol 中国队亚运会夺冠</Timeline.Item>
        <Timeline.Item>目前已回到国内备战s8</Timeline.Item>
        <Timeline.Item>接下来是LPL 季后赛</Timeline.Item>
        <Timeline.Item>目前 RNG, RW, IG 最后可能进入 S8</Timeline.Item>
      </Timeline>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-timeline classnames", () => {
    const wrapper = shallow(
      <Timeline>
        <Timeline.Item type="info">信息</Timeline.Item>
        <Timeline.Item type="error">错误</Timeline.Item>
        <Timeline.Item type="warning">警告</Timeline.Item>
        <Timeline.Item type="disabled">禁用</Timeline.Item>
        <Timeline.Item type="loading">EDG.Clearlove9 正在连接...</Timeline.Item>
      </Timeline>
    );
    assert(wrapper.find(".cuke-timeline").length === 1);
  });

  it("should render custom type", () => {
    const wrapper = render(
      <Timeline>
        <Timeline.Item type="info">信息</Timeline.Item>
        <Timeline.Item type="error">错误</Timeline.Item>
        <Timeline.Item type="warning">警告</Timeline.Item>
        <Timeline.Item type="disabled">禁用</Timeline.Item>
        <Timeline.Item type="loading">EDG.Clearlove9 正在连接...</Timeline.Item>
      </Timeline>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render custom color", () => {
    const wrapper = render(
      <Timeline>
        <Timeline.Item color="#654">三天前 lol 中国队亚运会夺冠</Timeline.Item>
        <Timeline.Item color="yellow">目前已回到国内备战s8</Timeline.Item>
        <Timeline.Item color="green">接下来是LPL 季后赛</Timeline.Item>
        <Timeline.Item color="#f63">
          目前 RNG, RW, IG 最后可能进入 S8
				</Timeline.Item>
      </Timeline>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render custom animate", () => {
    const wrapper = render(
      <div>
        <Timeline animate="slideLeft">
          <Timeline.Item>1</Timeline.Item>
          <Timeline.Item>2</Timeline.Item>
        </Timeline>
        <Timeline animate="slideRight">
          <Timeline.Item>1</Timeline.Item>
          <Timeline.Item>2</Timeline.Item>
        </Timeline>
        <Timeline animate="slideUp">
          <Timeline.Item>1</Timeline.Item>
          <Timeline.Item>2</Timeline.Item>
        </Timeline>
        <Timeline animate="slideBottom">
          <Timeline.Item>1</Timeline.Item>
          <Timeline.Item>2</Timeline.Item>
        </Timeline>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
