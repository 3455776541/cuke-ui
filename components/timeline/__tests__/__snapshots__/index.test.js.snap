// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline/> should render Timeline 1`] = `
<ul
  class="cuke-timeline"
>
  <li
    class="cuke-timeline-item cuke-timeline-item-primary"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0s"
    >
      三天前 lol 中国队亚运会夺冠
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-primary"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.1s"
    >
      目前已回到国内备战s8
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-primary"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.2s"
    >
      接下来是LPL 季后赛
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-primary"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.3s"
    >
      目前 RNG, RW, IG 最后可能进入 S8
    </div>
  </li>
</ul>
`;

exports[`<Timeline/> should render custom animate 1`] = `
<div>
  <ul
    class="cuke-timeline cuke-timeline-animated cuke-timeline-slideLeft"
  >
    <li
      class="cuke-timeline-item cuke-timeline-item-primary"
    >
      <div
        class="cuke-timeline-item-line"
      />
      <div
        class="cuke-timeline-item-dot"
        style="border-color:"
      >
        <div
          class="cuke-timeline-dot"
        />
      </div>
      <div
        class="cuke-timeline-item-content"
        style="animation-delay:0s"
      >
        1
      </div>
    </li>
    <li
      class="cuke-timeline-item cuke-timeline-item-primary"
    >
      <div
        class="cuke-timeline-item-line"
      />
      <div
        class="cuke-timeline-item-dot"
        style="border-color:"
      >
        <div
          class="cuke-timeline-dot"
        />
      </div>
      <div
        class="cuke-timeline-item-content"
        style="animation-delay:0.1s"
      >
        2
      </div>
    </li>
  </ul>
  <ul
    class="cuke-timeline cuke-timeline-animated cuke-timeline-slideRight"
  >
    <li
      class="cuke-timeline-item cuke-timeline-item-primary"
    >
      <div
        class="cuke-timeline-item-line"
      />
      <div
        class="cuke-timeline-item-dot"
        style="border-color:"
      >
        <div
          class="cuke-timeline-dot"
        />
      </div>
      <div
        class="cuke-timeline-item-content"
        style="animation-delay:0s"
      >
        1
      </div>
    </li>
    <li
      class="cuke-timeline-item cuke-timeline-item-primary"
    >
      <div
        class="cuke-timeline-item-line"
      />
      <div
        class="cuke-timeline-item-dot"
        style="border-color:"
      >
        <div
          class="cuke-timeline-dot"
        />
      </div>
      <div
        class="cuke-timeline-item-content"
        style="animation-delay:0.1s"
      >
        2
      </div>
    </li>
  </ul>
  <ul
    class="cuke-timeline cuke-timeline-animated cuke-timeline-slideUp"
  >
    <li
      class="cuke-timeline-item cuke-timeline-item-primary"
    >
      <div
        class="cuke-timeline-item-line"
      />
      <div
        class="cuke-timeline-item-dot"
        style="border-color:"
      >
        <div
          class="cuke-timeline-dot"
        />
      </div>
      <div
        class="cuke-timeline-item-content"
        style="animation-delay:0s"
      >
        1
      </div>
    </li>
    <li
      class="cuke-timeline-item cuke-timeline-item-primary"
    >
      <div
        class="cuke-timeline-item-line"
      />
      <div
        class="cuke-timeline-item-dot"
        style="border-color:"
      >
        <div
          class="cuke-timeline-dot"
        />
      </div>
      <div
        class="cuke-timeline-item-content"
        style="animation-delay:0.1s"
      >
        2
      </div>
    </li>
  </ul>
  <ul
    class="cuke-timeline cuke-timeline-animated cuke-timeline-slideBottom"
  >
    <li
      class="cuke-timeline-item cuke-timeline-item-primary"
    >
      <div
        class="cuke-timeline-item-line"
      />
      <div
        class="cuke-timeline-item-dot"
        style="border-color:"
      >
        <div
          class="cuke-timeline-dot"
        />
      </div>
      <div
        class="cuke-timeline-item-content"
        style="animation-delay:0s"
      >
        1
      </div>
    </li>
    <li
      class="cuke-timeline-item cuke-timeline-item-primary"
    >
      <div
        class="cuke-timeline-item-line"
      />
      <div
        class="cuke-timeline-item-dot"
        style="border-color:"
      >
        <div
          class="cuke-timeline-dot"
        />
      </div>
      <div
        class="cuke-timeline-item-content"
        style="animation-delay:0.1s"
      >
        2
      </div>
    </li>
  </ul>
</div>
`;

exports[`<Timeline/> should render custom color 1`] = `
<ul
  class="cuke-timeline"
>
  <li
    class="cuke-timeline-item cuke-timeline-item-primary"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:#654"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0s"
    >
      三天前 lol 中国队亚运会夺冠
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-primary"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:yellow"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.1s"
    >
      目前已回到国内备战s8
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-primary"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:green"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.2s"
    >
      接下来是LPL 季后赛
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-primary"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:#f63"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.3s"
    >
      目前 RNG, RW, IG 最后可能进入 S8
    </div>
  </li>
</ul>
`;

exports[`<Timeline/> should render custom type 1`] = `
<ul
  class="cuke-timeline"
>
  <li
    class="cuke-timeline-item cuke-timeline-item-info"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0s"
    >
      信息
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-error"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.1s"
    >
      错误
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-warning"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.2s"
    >
      警告
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-disabled"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot"
      style="border-color:"
    >
      <div
        class="cuke-timeline-dot"
      />
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.3s"
    >
      禁用
    </div>
  </li>
  <li
    class="cuke-timeline-item cuke-timeline-item-loading"
  >
    <div
      class="cuke-timeline-item-line"
    />
    <div
      class="cuke-timeline-item-dot-loading"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color:"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="6"
        />
        <line
          x1="12"
          x2="12"
          y1="18"
          y2="22"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="4.93"
          y2="7.76"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="16.24"
          y2="19.07"
        />
        <line
          x1="2"
          x2="6"
          y1="12"
          y2="12"
        />
        <line
          x1="18"
          x2="22"
          y1="12"
          y2="12"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="19.07"
          y2="16.24"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="7.76"
          y2="4.93"
        />
      </svg>
    </div>
    <div
      class="cuke-timeline-item-content"
      style="animation-delay:0.4s"
    >
      EDG.Clearlove9 正在连接...
    </div>
  </li>
</ul>
`;
